validator/client/attest.go

```diff
@@ -79,6 +73,32 @@ func (v *validator) SubmitAttestation(ctx context.Context, slot uint64, pubKey [
 		}
 		return
 	}
+	indexedAtt.Signature = sig
+	slashable, err := v.localSlashingProtector.IsSlashableAttestation(ctx, indexedAtt, pubKey, signingRoot)
+	if err != nil {
+		log.WithFields(
+			attestationLogFields(pubKey, indexedAtt),
+		).WithError(err).Error("Could not check attestation safety with slashing protection, not submitting")
+		return
+	}
+	if remoteProtector, ok := v.remoteSlashingProtector.(*remote.Service); ok && remoteProtector != nil {
+		slashable, err = v.remoteSlashingProtector.IsSlashableAttestation(ctx, indexedAtt, pubKey, signingRoot)
```

Comment:
This variable shadowing is not good! We are basically ignoring the results of the local slashing protection, if we leave this as is. 

Recommend: assign two different variables and check `if rSlashable || lSlashable {`